commit b845b6caac2b3826f707d5b62b9839c9e1a3c3d9 (refs/changes/45/1145/2)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-12-19T23:27:01+00:00 (2 years, 10 months ago)
    
    joyent/node-mname-client#4 crash due to use of this.gotoState in sockets.js
    Reviewed by: David Pacheco <dap@joyent.com>

diff --git a/lib/sockets.js b/lib/sockets.js
index a88d93d..4e1ce0b 100644
--- a/lib/sockets.js
+++ b/lib/sockets.js
@@ -77,7 +77,7 @@ DnsTcpSocket.prototype.cancel = function (req) {
 		delete (this.dts_requests[id]);
 		if (Object.keys(this.dts_requests).length < 1) {
 			if (this.dts_ending)
-				this.gotoState('closed');
+				this.emit('finalReqCancelled');
 			else
 				this.dts_socket.unref();
 		}
@@ -190,6 +190,9 @@ DnsTcpSocket.prototype.state_connected = function (S) {
 			S.gotoState('error');
 		}
 	});
+	S.on(this, 'finalReqCancelled', function () {
+		S.gotoState('closed');
+	});
 	S.immediate(function () {
 		self.emit('ready');
 	});
diff --git a/package.json b/package.json
index d455ae2..2c8c867 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "mname-client",
   "description": "DNS client library for node.js",
-  "version": "0.5.0",
+  "version": "0.5.1",
   "author": "Joyent, Inc",
   "scripts": {
     "test": "tape test/*.test.js"
